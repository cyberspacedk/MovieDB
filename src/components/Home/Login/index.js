import LoginForm from './container';

export default LoginForm;
