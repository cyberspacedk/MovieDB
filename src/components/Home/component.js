import React from 'react';
import PropTypes from 'prop-types';
import Dashboard from './Dashboard';
import Login from './Login';

const Home = ({ isAuthentificated }) => (
  <>{!isAuthentificated ? <Login /> : <Dashboard />}</>
);

Home.propTypes = {
  isAuthentificated: PropTypes.bool,
};
export default Home;
