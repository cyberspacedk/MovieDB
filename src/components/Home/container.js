import { connect } from 'react-redux';
import { isAuthentificated } from '../../store/authentifiction/selectors';
import Home from './component';

export const mstp = state => ({
  isAuthentificated: isAuthentificated(state),
});

export default connect(mstp)(Home);
