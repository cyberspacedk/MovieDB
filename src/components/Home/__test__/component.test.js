import React from 'react';
import { shallow } from 'enzyme';
import Home from '../component';

describe('Component: Home', () => {
  const props = {
    isAuthentificated: true,
  };

  it('Match its snapshot. Display Dashboard', () => {
    const container = shallow(<Home {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('Match its snapshot. Display LoginPage ', () => {
    const nextProps = {
      isAuthentificated: false,
    };
    const TopFilmsComponent = shallow(<Home {...nextProps} />);
    expect(TopFilmsComponent).toMatchSnapshot();
  });
});
