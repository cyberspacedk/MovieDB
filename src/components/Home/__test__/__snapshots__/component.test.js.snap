// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Home Match its snapshot. Display Dashboard 1`] = `
<Fragment>
  <Dashboard />
</Fragment>
`;

exports[`Component: Home Match its snapshot. Display LoginPage  1`] = `
<Fragment>
  <Connect(WithFormik(LoginForm)) />
</Fragment>
`;
